src/math_parser_diag.cpp

```diff
@@ -1317,6 +1317,56 @@ std::function<double( dialogue & )> vision_range_eval( char scope,
     };
 }
 
+std::function<double( dialogue & )> calories_eval( char scope,
+        std::vector<diag_value> const &/* params */, diag_kwargs const &kwargs )
+{
+    diag_value format_value( std::string( "raw" ) );
+    if( kwargs.count( "format" ) != 0 ) {
+        format_value = *kwargs.at( "format" );
+    }
+
+    return[format_value, beta = is_beta( scope )]( dialogue const & d ) {
+        std::string format = format_value.str( d );
+        if( format != "raw" && format != "percent" ) {
+            debugmsg( R"(Unknown format type "%s" for calories, assumning "raw")", format );
+            format = "raw";
+        }
+
+        if( format == "percent" ) {
+            if( d.actor( beta )->get_character() ) {
+                int divisor = d.actor( beta )->get_healthy_kcal() / 100;
+                //if no data, default to default height of 175cm
+                if( divisor == 0 ) {
+                    divisor = 118169 / 100;
+                }
+                return d.actor( beta )->get_stored_kcal() / divisor;
+            } else {
+                debugmsg( "Percent can be used only with character" );
+                return 0;
+            }
+        } else if( format == "raw" ) {
+            if( d.actor( beta )->get_character() ) {
+                return d.actor( beta )->get_stored_kcal();
+            }
+            item_location *it = d.actor( beta )->get_item();
+            if( it && *it ) {
+                npc dummy;
+                return dummy.compute_effective_nutrients( *it->get_item() ).kcal();
+            }
+        } else {
+            debugmsg( "For calories(), talker is not character nor item" );
+            return 0;
+        };
```

Comment:
```suggestion
        }
        
        debugmsg( "For calories(), talker is not character nor item" );
        return 0;
```

This should fix the issue. If it's wrapped in an `else`, it won't catch paths like when `if( it && *it )` was false.